components/shopify/actions/common/metafield-actions.mjs

```diff
@@ -0,0 +1,139 @@
+import shopify from "../../shopify.app.mjs";
+import consts from "../common/consts.mjs";
+
+export default {
+  props: {
+    shopify,
+    ownerResource: {
+      type: "string",
+      label: "Resource Type",
+      description: "Filter by the resource type on which the metafield is attached to",
+      options: consts.RESOURCE_TYPES,
+      reloadProps: true,
+    },
+  },
+  methods: {
+    async getOwnerIdProp(ownerResource) {
+      const product = {
+        ...shopify.propDefinitions.productId,
+        options: async ({ prevContext }) => {
+          return this.shopify.getProductOptions(prevContext);
+        },
+      };
+      const productVariant = {
+        ...shopify.propDefinitions.productVariantId,
+        options: async ({ prevContext }) => {
+          return this.shopify.getProductVariantOptions(this.productId, prevContext);
+        },
+      };
+      const productImage = {
+        ...shopify.propDefinitions.imageId,
+        options: async() => {
+          return this.shopify.getImageOptions(this.productId);
+        },
+        optional: false,
+      };
+      const customer = {
+        ...shopify.propDefinitions.customerId,
+        options: async ({
+          prevContext, query,
+        }) => {
+          return this.shopify.getCustomerOptions(prevContext, query);
+        },
+      };
+      const collection = {
+        ...shopify.propDefinitions.collectionId,
+        options: async () => {
+          return this.shopify.getCollectionOptions();
+        },
+        optional: false,
+      };
+      const blog = {
+        ...shopify.propDefinitions.blogId,
+        options: async () => {
+          return this.shopify.getBlogOptions();
+        },
+      };
+      const article = {
+        ...shopify.propDefinitions.articleId,
+        options: async () => {
+          return this.shopify.getArticleOptions(this.blogId);
+        },
+      };
+      const page = {
+        ...shopify.propDefinitions.pageId,
+        options: async () => {
+          return this.shopify.getPageOptions();
+        },
+      };
+      const order = {
+        ...shopify.propDefinitions.orderId,
+        options: async () => {
+          return this.shopify.getOrderOptions();
+        },
+      };
+      const draftOrder = {
+        ...shopify.propDefinitions.draftOrderId,
+        options: async () => {
+          return this.shopify.getDraftOrderOptions();
+        },
+      };
+
+      let props = {};
+      if (ownerResource === "product") {
```

Comment:
Looks like we always have `ownerId` field in the props. If that's the case, can we start `props` with this property and just remove some of these `ifs`?